tools/Runtime Condenser/condenser.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# Runtime Condenser, rewritten in python by tkdrg
+# Only the first set of proc, file, usr and src will be kept. The rest is truncated.
+# This script is valid on both python2 and python3
+runtimes = dict()
+infloops = dict()
+harddels = dict()
+
+
+def readFromFile(target='input.txt'):
+	global runtimes, infloops, harddels
+	f = open(target)
+
+	for line in iter(f.readline, ''):
```

Comment:
Does that also take in account any further updates to the cursor? I was afraid that for line in f would break due to the f.readline()s inside the loop (which are expected to throw away the lines).